test/e2e/storage/csi_volumes.go

```diff
@@ -367,57 +369,96 @@ func initCSIgcePD(f *framework.Framework, config framework.VolumeTestConfig) csi
 		claimSize:    "5Gi",
 		expectedSize: "5Gi",
-		nodeName:     node.Name,
+		nodeName:     g.config.ServerNodeName,
 	}
 }
 
 func (g *gcePDCSIDriver) createCSIDriver() {
 	By("deploying gce-pd driver")
-	f := g.f
-	cs := f.ClientSet
-	config := g.config
-	g.controllerServiceAccount = csiServiceAccount(cs, config, "gce-controller", false /* teardown */)
-	g.nodeServiceAccount = csiServiceAccount(cs, config, "gce-node", false /* teardown */)
-	csiClusterRoleBindings(cs, config, false /* teardown */, g.controllerServiceAccount, g.controllerClusterRoles)
-	csiClusterRoleBindings(cs, config, false /* teardown */, g.nodeServiceAccount, g.nodeClusterRoles)
-	utils.PrivilegedTestPSPClusterRoleBinding(cs, config.Namespace, false, /* teardown */
-		[]string{g.controllerServiceAccount.Name, g.nodeServiceAccount.Name})
-	role := csiControllerRole(cs, config, false)
-	csiControllerRoleBinding(cs, config, false, role, g.controllerServiceAccount)
-	deployGCEPDCSIDriver(cs, config, false /* teardown */, f, g.nodeServiceAccount, g.controllerServiceAccount)
+	cleanup, err := g.f.CreateFromManifest(func(object interface{}) {
+		patchCSIDeployment(g.f, "com.google.csi.gcepd", "gce-driver", "csi-external-provisioner", g.config.ServerNodeName, object)
+	},
+		"test/e2e/testing-manifests/storage-csi/driver-registrar/rbac.yaml",
+		"test/e2e/testing-manifests/storage-csi/external-attacher/rbac.yaml",
+		"test/e2e/testing-manifests/storage-csi/external-provisioner/rbac.yaml",
+		"test/e2e/testing-manifests/storage-csi/gce-pd/csi-controller-rbac.yaml",
+		"test/e2e/testing-manifests/storage-csi/gce-pd/node_ds.yaml",
+		"test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml",
+		"test/e2e/testing-manifests/storage-csi/gce-pd/controller_service.yaml",
+	)
+	g.cleanup = cleanup
+	if err != nil {
+		framework.Failf("deploying csi hostpath driver: %v", err)
+	}
 }
 
 func (g *gcePDCSIDriver) cleanupCSIDriver() {
 	By("uninstalling gce-pd driver")
-	f := g.f
-	cs := f.ClientSet
-	config := g.config
-	deployGCEPDCSIDriver(cs, config, true /* teardown */, f, g.nodeServiceAccount, g.controllerServiceAccount)
-	csiClusterRoleBindings(cs, config, true /* teardown */, g.controllerServiceAccount, g.controllerClusterRoles)
-	csiClusterRoleBindings(cs, config, true /* teardown */, g.nodeServiceAccount, g.nodeClusterRoles)
-	utils.PrivilegedTestPSPClusterRoleBinding(cs, config.Namespace, true, /* teardown */
-		[]string{g.controllerServiceAccount.Name, g.nodeServiceAccount.Name})
-	role := csiControllerRole(cs, config, true)
-	csiControllerRoleBinding(cs, config, true, role, g.controllerServiceAccount)
-	csiServiceAccount(cs, config, "gce-controller", true /* teardown */)
-	csiServiceAccount(cs, config, "gce-node", true /* teardown */)
+	if g.cleanup != nil {
+		g.cleanup()
+	}
+}
+
+// TODO (?): the storage.csi.image.version and storage.csi.image.registry
+// settings are ignored. We could patch the image definitions.
+// TODO: way too many parameters...
```

Comment:
I wonder if the selector and labels also need to be patched?
https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/deploy/kubernetes/stable/controller.yaml#L8

Reply:
And the CSI_ADDRESS? https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/deploy/kubernetes/stable/node.yaml#L26

Reply:
Selector and labels do not need to be patched because they are relative to the namespace, which is unique to the test. Not renaming when it isn't needed makes debugging easier.

`CSI_ADDRESS` doesn't need to be patched in the manifest used by the E2E testing because it uses `/csi/csi.sock` as fixed path:
https://github.com/kubernetes/kubernetes/blob/689df2010da51b19fec2cdfe3297533cb35c90bd/test/e2e/testing-manifests/storage-csi/gce-pd/node_ds.yaml#L21

IMHO using that full path in `node.yaml` is unnecessarily complex. But perhaps I'm just not aware of some advantage that the full path has?

Apropos that line, what's the advantage of defining an env variable with a fixed value and then referencing that value? That's again something were I don't see the advantage over just `--csi-address=/csi/csi.sock`.